server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -1177,6 +1167,23 @@ private UserVm upgradeStoppedVirtualMachine(Long vmId, Long svcOffId, Map<String
 
     }
 
+    private void resizeVolumeWithNewOfferings(VMInstanceVO vmInstance, ServiceOfferingVO newServiceOffering)
+            throws ResourceAllocationException {
+        DiskOfferingVO newROOTDiskOffering = _diskOfferingDao.findById(newServiceOffering.getId());
+
+        List<VolumeVO> vols = _volsDao.findReadyRootVolumesByInstance(vmInstance.getId());
+
+        for (final VolumeVO rootVolumeOfVm : vols) {
+            rootVolumeOfVm.setDiskOfferingId(newROOTDiskOffering.getId());
+
+            _volsDao.update(rootVolumeOfVm.getId(), rootVolumeOfVm);
+
+            ResizeVolumeCmd resizeVolumeCmd = new ResizeVolumeCmd(rootVolumeOfVm.getId(), newROOTDiskOffering.getMinIops(), newROOTDiskOffering.getMaxIops());
+
+            _volumeService.resizeVolume(resizeVolumeCmd);
```

Comment:
@slavkap In case any failure in the ResizeVolumeCmd execution, the volume will still point to the new offering. So, better to update the new offering in the volume after ResizeVolumeCmd execution is successful.

Reply:
thanks @sureshanaparti for the code review! I will move the update of new disk offerings after the resizeVolume, but maybe I need to move the method invocation of resizeVolumeWithNewOfferings before upgradeVMDb (like how it's invoked in upgradeStoppedVirtualMachine)?